### PR TITLE
Fix #5133  Audio language name showing two times

### LINKED ISSUE #5133: [BUG][Blocked]:  Audio language name  showing two times
### Describe the bug

Audio language shows two times English, But  one should display Spanish

### Steps To Reproduce

1) go toHome
2) Select Ratios and proportion Topic
3) click on  "what is a Ratio ?"
4) click on continue
5) tap on audio
6) select Audio language

### Expected Behavior

Audio language list should display hindi, English, and  Spanish.

### Screenshots/Videos

WhatsApp Image 2023-08-14 at 10 27 08 AM (1)

### What device/emulator are you using?

Pixel XL(emulator)

### Which Android version is your device/emulator running?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Vishwajith-Shettigar**
## Explanation
Fix #5133, filtered the language codes to display only the languages supported by Oppia.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

| Before fix|
|--------|
| WhatsApp Image 2023-08-14 at 10 27 08 AM (1) | 

|After fix|
|--------|
| languageafetr | 

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes , PTAL

File: app/src/main/java/org/oppia/android/app/player/audio/AudioViewModel.kt [hunk 1]
```diff
@@ -123,6 +130,31 @@ class AudioViewModel @Inject constructor(
...
+  }
+
+  fun getSupportedLanguageCodes(): MutableList<String> {
+    val supportedAudioLanguageCodes = AudioLanguage.values().toList()
+      .map { audioLanguage ->
+        getAudioLanguage(audioLanguage)
+      } as MutableList<String>
+    supportedAudioLanguageCodes.add("hi-en")
+    return supportedAudioLanguageCodes
+  }
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/audio/AudioViewModel.kt]: Could be flattened:
```suggestion
  fun getSupportedLanguageCodes(): List<String> {
    val supportedAudioLanguageCodes = AudioLanguage.values()
      .map { audioLanguage ->
        getAudioLanguage(audioLanguage)
      }
      .toMutableList()
    supportedAudioLanguageCodes.add("hi-en")
    return supportedAudioLanguageCodes
  }
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! 

1. Could you please add before and after screenshots?
2.  We currently do not have tests for `LanguageDialogFragment`. Do you think you can explore adding one, and testing this fix? See: https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing

**Comment by adhiamboperes** [comment]
@Vishwajith-Shettigar, I'm closing this PR since the desired implementation is blocked. See https://github.com/oppia/oppia-android/issues/5133#issuecomment-1839602314

**Pull request closed without merging**